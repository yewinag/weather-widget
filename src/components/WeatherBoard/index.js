import React, { Component } from 'react';
import styled from 'styled-components';

const LoadingLayout = styled.div`
    display: 'flex';
    align-items: 'center';
`
const WeatherLayout = styled.div`
    min-height: 400px;
    border: 1px solid #ddd;
    padding: 20px;
`
const Header = styled.h4`
    font-weight: bold;
    font-size: 24px;
    text-align: center;
`

const WeatherTempLayout = styled.div`
    display: flex;
    justify-content: space-between;
`

const Text = styled.p`
    font-size: 18px;
`

const ForecastLayout = styled.div`
    display: flex;
    justify-content: space-between;
`
const ForecastItem = styled.div`
    padding: 10;
    align-items: center;
    font-weight: bold;
`
const Layout = styled.div`

`
class WeatherBoard extends Component {

    renderLoading = () => (        
        <LoadingLayout>loading..</LoadingLayout>
    )
    renderCityWeather = (weather) => (
        <Layout>
            <Header>{weather.weather.name}</Header>
            <WeatherTempLayout>
                <Text>{weather.weather.main.temp}</Text>
                <div>
                    <Text>
                        winds: {weather.weather.wind.speed}
                    </Text>
                    <Text>
                        Sunrise: {weather.weather.sys.sunrise}
                    </Text>
                    <Text>
                        Sunset: {weather.weather.sys.sunset}
                    </Text>
                </div>
            </WeatherTempLayout>
        </Layout>
    )
    renderError = (weather) => (
        <div>
            {weather.weather.message}
        </div>
    )
    renderForecast = (weather) => (
        <ForecastLayout>
            {
                weather.forecast.daily.map((day, index) => (
                    <ForecastItem key={index}>
                        {day.temp.day}
                    </ForecastItem>
                ))
            }
        </ForecastLayout>
    )
    render() {
        const { weather } = this.props;
        return (
            <WeatherLayout>
                {weather.weather.isFetching && this.renderLoading()}
                {weather.weather.cod === 200 && this.renderCityWeather(weather)}
                {weather.weather.cod == 404 && this.renderError(weather)}
                {weather.weather.cod == 400 && this.renderError(weather)}
                {weather.forecast.hasOwnProperty('daily') && this.renderForecast(weather)}                                                        
            </WeatherLayout>
        );
    }
}

export default WeatherBoard;