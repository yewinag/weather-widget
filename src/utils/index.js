
import { applyMiddleware, createStore } from 'redux';
import rootReducer from '../reducers';
import { middlewares } from '../store';

export const testStore = (initialState) => {
    const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);
    return createStoreWithMiddleware(rootReducer, initialState);
};