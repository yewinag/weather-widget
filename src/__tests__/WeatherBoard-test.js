import React from 'react';
import { Provider } from 'react-redux';
import { testStore } from '../utils';
import WeatherBoard from '../components/WeatherBoard';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });

describe('weather board', () => {
    let store;
    beforeEach(() => {
        store = testStore();
    });   
    const testProps = {
        weather: {},
        forecast: {}
    }
    it('render weather board', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <WeatherBoard weather={testProps}/>                
            </Provider>
        )
        expect(wrapper.exists()).toBe(true);
        expect(wrapper.props().children.props.weather).toBe(testProps);
        expect(wrapper).toMatchSnapshot();
    });
});