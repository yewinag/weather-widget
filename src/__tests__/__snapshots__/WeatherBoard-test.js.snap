// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`weather board render weather board 1`] = `ShallowWrapper {}`;
